### PR TITLE
tabby: 0.11.0 -> 0.14.0

### PR DESCRIPTION
**Pull request by ghthor**
## Description of changes

1. remove tabby scheduler from the nixos module
    1. this functionality is now available from the tabby web interface
1. replace static link of llama-cpp w/ external binary
    1. tabby now supports orchestrating llama-cpp via external process
1. upgrade tabby to v0.14

I will follow up with another PR to upgrade tabby to the latest released version
as 0.14 is currently out of date.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ocfox** [comment]
0.16.1 released and could build success.

**Comment by Ramblurr** [comment]
0.17.0 is out, and I got it building, see https://github.com/Ramblurr/nixpkgs/commit/e89e73d1005a7e4ad2e419b1a69159ac76870078

**Comment by ghthor** [comment]
> 0.17.0 is out, and I got it building, see [Ramblurr@e89e73d](https://github.com/Ramblurr/nixpkgs/commit/e89e73d1005a7e4ad2e419b1a69159ac76870078)

Pushed upgrade to 0.17.0 to a separate PR

1. https://github.com/NixOS/nixpkgs/pull/345724

**Comment by Ramblurr** [comment]
Hmm that failing check is problematic. Got to be all green before merge.

**Comment by Ramblurr** [comment]
It could be yea. You can just include the lock file in the src tree, that's pretty common for other packages.

File: nixos/modules/services/misc/tabby.nix [hunk 1]
```diff
@@ -80,29 +88,6 @@ in
         '';
       };
 
-      settings = lib.mkOption {
```
Comment (SuperSandro2000) [inline nixos/modules/services/misc/tabby.nix:83]: Please add a lib.mkRemovedOptionModule

File: nixos/modules/services/misc/tabby.nix [hunk 2]
```diff
@@ -113,34 +98,16 @@ in
           > https://tabby.tabbyml.com/docs/configuration#usage-collection
         '';
       };
-
-      indexInterval = lib.mkOption {
```
Comment (SuperSandro2000) [inline nixos/modules/services/misc/tabby.nix:117]: same here

File: pkgs/by-name/ta/tabby/package.nix [hunk 3]
```diff
@@ -66,70 +71,71 @@ let
...
-  cudaBuildInputs = [ llamaccpPackage ];
-  rocmBuildInputs = [ llamaccpPackage ];
+  # TODO(ghthor): support linux aarch64
+  buildTarget = if stdenv.isDarwin then "aarch64-apple-darwin" else "x86_64-unknown-linux-gnu";
 
 in
 rustPlatform.buildRustPackage {
   inherit pname version;
   inherit featureDevice;
+  inherit buildTarget;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ta/tabby/package.nix:106]:
```suggestion
  inherit pname version featureDevice buildTarget;
```

File: pkgs/by-name/ta/tabby/package.nix [hunk 4]
```diff
@@ -66,70 +71,71 @@ let
...
 
   src = fetchFromGitHub {
     owner = "TabbyML";
     repo = "tabby";
     rev = "v${version}";
-    hash = "sha256-OgAE526aW3mVqf6fVmBmL5/B4gH9B54QLEITQk9Kgsg=";
-    fetchSubmodules = true;
+    hash = "sha256-NA+Shkn+m5yZh9kOkvt+280JvFykReMrz7R0OP5x3AA=";
+    fetchSubmodules = false;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ta/tabby/package.nix:113]:
```suggestion
```
that's the default

File: pkgs/by-name/ta/tabby/package.nix [hunk 5]
```diff
@@ -66,70 +71,71 @@ let
...
   };
 
   cargoLock = {
-    lockFile = ./Cargo.lock;
+    lockFile = fetchurl {
+      url = "https://raw.githubusercontent.com/TabbyML/tabby/v${version}/Cargo.lock";
```
Comment (SuperSandro2000) [inline pkgs/by-name/ta/tabby/package.nix:118]: That is causing IFD. You must copy the file to the repo where it was previously and change this line back.

File: pkgs/by-name/ta/tabby/package.nix [hunk 6]
```diff
@@ -138,32 +144,33 @@ rustPlatform.buildRustPackage {
...
-  patches = [ ./0001-nix-build-use-nix-native-llama-cpp-package.patch ];
-
   # Fails with:
   # file cannot create directory: /var/empty/local/lib64/cmake/Llama
   doCheck = false;
 
   passthru.updateScript = nix-update-script { };
 
+  patches = [ ];
+
```
Comment (SuperSandro2000) [inline pkgs/by-name/ta/tabby/package.nix:161]:
```suggestion
```

**Comment by wegank** [comment]
Done in #350323.

**Pull request closed without merging**